llvm/include/llvm/Transforms/Utils/CodeExtractor.h

```diff
@@ -101,12 +102,25 @@ class CodeExtractorAnalysisCache {
 
     // Bits of intermediate state computed at various phases of extraction.
     SetVector<BasicBlock *> Blocks;
-    unsigned NumExitBlocks = std::numeric_limits<unsigned>::max();
-    Type *RetTy;
 
-    // Mapping from the original exit blocks, to the new blocks inside
-    // the function.
-    SmallVector<BasicBlock *, 4> OldTargets;
+    /// Lists of blocks that are branched from the code region to be extracted,
+    /// also called the exit blocks. Each block is contained at most once. Its
+    /// order defines the return value of the extracted function.
+    ///
+    /// When there is just one (or no) exit block, the return value is
+    /// irrelevant.
+    ///
+    /// When there are exactly two exit blocks, the extracted function returns a
+    /// boolean. For SwitchCases[0], it returns 'true'. For SwitchCases[1] it
+    /// returns 'false'.
+    /// NOTE: Since a boolean is represented by i1, SwitchCases[0] returns 1 and
+    ///       SwitchCases[1] returns 0, which is inverted with the the regular
+    ///       switch case below.
+    ///
+    /// When there are 3 or more exit blocks, leaving the extracted function via
+    /// the first block it returns 0. When leaving via the second entry it
+    /// returns 1, etc.
+    SmallVector<BasicBlock *> SwitchCases;
```

Comment:
nit: I think this should be "ExitBlocks"

Reply:
I intentionally named it `SwitchCases` to indicate that it is not just an unordered list of exist blocks, but the order is used to determine the extracted function's return values and the replacement switch statement. I am not married to the name (could also be `ExtractedFunctionReturnValues`), but I honestly think that `ExitBlocks` would be a misleading name.

Despite this, do you still prefer `ExitBlocks`?

Reply:
Just any name that doesn't imply it's only used by a switch instruction.